src/js/components/TabContent.tsx

```diff
@@ -45,6 +46,7 @@ export default function TabContent() {
     )
 
   if (wsStatus === "disconnected") return <ConnectionError workspace={ws} />
+  if (wsStatus === "login-required") return <Login ws={ws.serialize()} />
```

Comment:
Why serialize the workspace here?

Reply:
I just checked the contents of `<Login />` and it doesn't look like there is any global dispatch calls that pass in the `ws` object. It looks like this is the only place `<Login />` is used as well. Changing the Login props to:

```diff
 type Props = {
-  ws: Workspace
+  ws: BrimWorkspace
 }
```

and then passing in `ws` seems to work fine with the compiler.
```diff
-  if (wsStatus === "login-required") return <Login ws={ws.serialize()} />
+  if (wsStatus === "login-required") return <Login ws={ws} />
```

If you were to need to serialize the object, it'd be best to do it at the last minute it's required. Not calling serialize also allows the ws={ws} prop to keep the same object references if it doesn't change. If later you made <Login /> a memo'd component, that would help avoid re-renders as well. Calling .serialize() returns a new object every time.

Reply:
Ah my bad, I had originally written `Workspaces.setWorkspaceToken` to take the whole workspace and hit the problem I described, began serializing workspace to fix, and then realized we really didn't need the whole workspace in that action and just passed id in. Thanks for pointing that out and I agree about the serialization strategy suggest.